src/Tizen.Log/Tizen/Log.cs

```diff
@@ -112,17 +114,52 @@ public static void Fatal(string tag, string message, [CallerFilePath] string fil
             Print(Interop.Dlog.LogPriority.DLOG_FATAL, tag, message, file, func, line);
         }
 
-        static void Print(Interop.Dlog.LogPriority priority, string tag, string message, string file, string func, int line)
+        static unsafe void Print(Interop.Dlog.LogPriority priority, string tag, string message, string file, string func, int line)
         {
-            if (String.IsNullOrEmpty(file))
-            {
-                Interop.Dlog.Print(priority, tag, "%s", message);
-            }
-            else
+            int length = tag.Length;
+            Span<byte> tagBuffer = length < 1024 ? stackalloc byte[length + 1] : new byte[length + 1];
+
+            length = message.Length;
+            Span<byte> messageBuffer = length < 1024 ? stackalloc byte[length + 1] : new byte[length + 1];
```

Comment:
byte size will be wrong if including CJK languages
you must use `Encoding.UTF8.GetMaxByteCount` 
https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.getmaxbytecount?view=net-7.0

Reply:
done